sw/host/tests/chip/gpio/src/gpio_pinmux.rs

```diff
@@ -430,18 +328,113 @@ fn test_pad_inversion(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
     Ok(())
 }
 
+fn test_opendrain(transport: &TransportWrapper, config: &Config) -> Result<()> {
+    let uart = transport.uart("console")?;
+
+    log::info!("Configuring debugger GPIOs as pulled-up inputs");
+    // The inputs (with respect to pinmux config) correspond to the output pins on the debug board.
+    let opendrain_inputs = config.input.values();
+    for pin in opendrain_inputs {
+        let pin = transport.gpio_pin(&pin.to_string())?;
+        pin.set_mode(PinMode::Input)?;
+        pin.set_pull_mode(PullMode::PullUp)?;
```

Comment:
(Repeated in `test_virtual_opendrain`): it is probably good practice to use `pin.set()` instead of individually calling `pin.set_mode()` and `pin.set_pull_mode()` as transports can override `pin.set()` to implement multiset (atomically set multiple pin attrs in one command to avoid glitching lines). HyperDebug for example [does this](https://github.com/lowRISC/opentitan/blob/5461e7af3dd1d7f534bf717e123cfb83bd9dbb10/sw/host/opentitanlib/src/transport/hyperdebug/gpio.rs#L106-L152).

```suggestion
        let pin = transport.gpio_pin(&pin.to_string())?;
        pin.set(Some(PinMode::Input), None, Some(PullMode::PullUp), None)?;
```